starknet_py/tests/e2e/conftest.py

```diff
@@ -467,3 +473,84 @@ async def cairo_serializer(gateway_account_client: AccountClient) -> CairoSerial
     contract = deployment_result.deployed_contract
 
     return CairoSerializer(identifier_manager=contract.data.identifier_manager)
+
+
+@pytest.fixture(scope="module")
+def fee_contract(pytestconfig, new_gateway_account_client: AccountClient) -> Contract:
+    abi = [
+        {
+            "inputs": [
+                {"name": "recipient", "type": "felt"},
+                {"name": "amount", "type": "Uint256"},
+            ],
+            "name": "transfer",
+            "outputs": [{"name": "success", "type": "felt"}],
+            "type": "function",
+        },
+        {
+            "members": [
+                {"name": "low", "offset": 0, "type": "felt"},
+                {"name": "high", "offset": 1, "type": "felt"},
+            ],
+            "name": "Uint256",
+            "size": 2,
+            "type": "struct",
+        },
+    ]
+
+    address = (
+        FEE_CONTRACT_ADDRESS
+        if pytestconfig.getoption("--net") != "devnet"
+        else DEVNET_FEE_CONTRACT_ADDRESS
+    )
+
+    return Contract(
+        address=address,
+        abi=abi,
+        client=new_gateway_account_client,
+    )
+
+
+@pytest_asyncio.fixture(scope="module")
+async def account_with_validate_deploy_class_hash(
+    new_gateway_account_client: AccountClient,
+) -> int:
+    compiled_contract = Compiler(
+        contract_source=(
+            contracts_dir / "account_with_validate_deploy.cairo"
+        ).read_text("utf-8"),
+        is_account_contract=True,
+    ).compile_contract()
+
+    declare_tx = await new_gateway_account_client.sign_declare_transaction(
+        compiled_contract=compiled_contract,
+        max_fee=MAX_FEE,
+    )
+    resp = await new_gateway_account_client.declare(transaction=declare_tx)
+    await new_gateway_account_client.wait_for_tx(resp.transaction_hash)
+
+    return resp.class_hash
```

Comment:
We could probably reuse `starknet.py/starknet_py/tests/e2e/client/fixtures/transactions.py`

Reply:
Unfortunately, I don't think so. I am returning only `class_hah` here and in the `fixtures/transactions.py` entire `DeployAccount` transaction is returned